src/OpenTelemetry.Exporter.Prometheus/Implementation/PrometheusExporterMetricsHttpServer.cs

```diff
@@ -128,13 +128,49 @@ private void WorkerThread()
                     ctxTask.Wait(this.tokenSource.Token);
                     var ctx = ctxTask.Result;
 
-                    if (!this.exporter.TryEnterSemaphore())
+                    try
```

Comment:
If the server is being called concurrently, only one thread can do the actual work, other threads will return the previous calculated response.